engine/src/rl/selfplay.cpp

```diff
@@ -78,13 +72,34 @@ SelfPlay::SelfPlay(RawNetAgent* rawAgent, MCTSAgent* mctsAgent, SearchLimits* se
     gamePGN.event = "SelfPlay";
     gamePGN.site = "Darmstadt, GER";
     gamePGN.round = "?";
-    gamePGN.is960 = false;
+    gamePGN.is960 = is960;
     this->exporter = new TrainDataExporter(string("data_") + mctsAgent->get_device_name() + string(".zarr"),
                                            rlSettings->numberChunks, rlSettings->chunkSize);
     filenamePGNSelfplay = string("games_") + mctsAgent->get_device_name() + string(".pgn");
     filenamePGNArena = string("arena_games_")+ mctsAgent->get_device_name() + string(".pgn");
     fileNameGameIdx = string("gameIdx_") + mctsAgent->get_device_name() + string(".txt");
 
+    // TODO: Faster and shorter. Boost.
+    ifstream f(filenamePGNSelfplay.c_str());
+    bool exist1 = f.good();
+    f.close();
+    if (exist1) {
+        const int delete1 = std::remove(filenamePGNSelfplay.c_str());
+        if (delete1 != 0) {
+            throw std::runtime_error("Could not delete " + filenamePGNSelfplay + " although it exists.");
+        }
+    }
+
+    ifstream f2(fileNameGameIdx.c_str());
+    bool exist2 = f2.good();
+    f2.close();
+    if (exist2) {
+        const int delete2 = std::remove(fileNameGameIdx.c_str());
+        if (delete2 != 0) {
+            throw std::runtime_error("Could not delete " + fileNameGameIdx + " although it exists.");
+        }
+    }
+
```

Comment:
You want to delete the previous pgn file here, right?
I think it would easier to simply open the pgn file in `writing` mode instead of `append` mode.

Reply:
Good idea! I have pushed the change.